pkg/apis/core/validation/validation.go

```diff
@@ -4197,12 +4236,47 @@ func ValidateNode(node *core.Node) field.ErrorList {
 	// That said, if specified, we need to ensure they are valid.
 	allErrs = append(allErrs, ValidateNodeResources(node)...)
 
-	if len(node.Spec.PodCIDR) != 0 {
-		_, err := ValidateCIDR(node.Spec.PodCIDR)
+	podCIDRsField := field.NewPath("spec", "podCIDRs")
+	// validate PodCIDRS only if we need to
+	if len(node.Spec.PodCIDRs) == 0 {
+		return allErrs
+	}
+
+	// all PodCIDRs should be valid ones
+	for idx, value := range node.Spec.PodCIDRs {
+		_, err := ValidateCIDR(value)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(podCIDRsField, node.Spec.PodCIDRs, fmt.Sprintf("not a valid CIDR[%v]=%v", idx, value)))
```

Comment:
All my comments above apply here, so I will not repeat them.  Specifically:  indexed fieldpaths, most appropriate error types, message strings